activation/activation.go

```diff
@@ -88,8 +87,8 @@ type Config struct {
 // it is responsible for initializing post, receiving poet proof and orchestrating nipst. after which it will
 // calculate total weight and providing relevant view as proof.
 type Builder struct {
-	pendingPoetClient atomic.UnsafePointer
-	started           atomic.Bool
+	pendingPoetClients chan []PoetProvingServiceClient
```

Comment:
The channel is not initialized anywhere.
TODO:
- [ ] fix
- [ ] write unit test

Reply:
what was the problem with atomic? i think it simpler that way

Reply:
Yeah, initially I thought that using a channel will simplify things, but it actually causes a few problems I had not anticipated. I will revert back to the atomic pointer.